build.sbt

```
@@ -1,22 +1,48 @@
-import Dependencies._
+import Dependencies.*
+
+// Your next release will be binary compatible with the previous one,
+// but it may not be source compatible (ie, it will be a minor release).
+ThisBuild / versionPolicyIntention := Compatibility.BinaryCompatible
+
+//TODO: after 3.1.0 release - clear up versionPolicyIgnored section
+/*
+versionPolicyReportDependencyIssues ignored dependencies when compared to safe-akka 3.0.0.
+All of those should not affect the library users, binary compatibility should be preserved.
+ */
+ThisBuild / versionPolicyIgnored ++= Seq(
+  //com.evolutiongaming:executor-tools_2.13: missing dependency
+  "com.evolutiongaming" %% "executor-tools",
+  //com.google.protobuf:protobuf-java: missing dependency
+  "com.google.protobuf" % "protobuf-java",
+  //org.scala-lang.modules:scala-java8-compat_2.13:
+  //  incompatible version change from 0.9.0 to 1.0.0 (compatibility: early semantic versioning
+  "org.scala-lang.modules" %% "scala-java8-compat",
+  //org.typelevel:cats-macros_2.13: missing dependency
+  "org.typelevel" %% "cats-macros",
+)
 
 lazy val commonSettings = Seq(
   organization := "com.evolutiongaming",
-  homepage := Some(url("http://github.com/evolution-gaming/safe-akka")),
+  homepage := Some(url("https://github.com/evolution-gaming/safe-akka")),
   startYear := Some(2017),
   organizationName := "Evolution",
-  organizationHomepage := Some(url("http://evolution.com")),
+  organizationHomepage := Some(url("https://evolution.com")),
   scalaVersion := crossScalaVersions.value.head,
-  crossScalaVersions := Seq("2.13.4", "2.12.12"),
+  crossScalaVersions := Seq("2.13.14", "3.3.3"),
+  Compile / scalacOptions ++= {
+    if (scalaBinaryVersion.value == "2.13") {
+      Seq(
+        "-Xsource:3"
+      )
+    } else Seq.empty
```

our plugin for `scalac` doesn't add any options for 3.3.3 😞 

maybe we can add some manually here (or upgrade our plugin with good and safe set for Scala 3.3.3), like:
```suggestion
    } else Seq("-deprecation")
```

---

we should add sbt-tpolecat instead. Not sure if it's worth doing in scope of this PR though

---

probably, just have to review it and either port the changes to our one or switch all our OSS to it

---

I'd rather leave it to the Evolution scalac opts plugin:
https://github.com/evolution-gaming/sbt-scalac-opts-plugin/issues/9

The projects will get the options with the plugin update.